File: app/src/main/java/de/dali/demonstrator/logging/SQLite/Dao/ImageDao.java
package de.dali.demonstrator.logging.SQLite.Dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import de.dali.demonstrator.logging.SQLite.Entity.Image;

@Dao
public interface ImageDao {

    @Query("Select * from image")
    List<Image> getAll();

    @Query("Delete from image")
    void deleteAll();

    //@Query("Select * from image where filename = :name")
    //List<Image> getImageByName(String name);

    @Query("Select * from image where imageID = :ID")
    List<Image> getImageByID(long ID);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertImage(Image image);

    @Update
    void updateImage(Image image);

    @Delete
    void deleteImage(Image image);
}

File: app/src/main/java/de/dali/demonstrator/data/local/dao/ImageDao.kt
package de.dali.demonstrator.data.local.dao

import androidx.room.*
import de.dali.demonstrator.data.local.entity.ImageEntity
import io.reactivex.Single

@Dao
interface ImageDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(image: ImageEntity): Long

    @Update
    fun update(image: ImageEntity)

    @Delete
    fun delete(image: ImageEntity)

    @Query("DELETE FROM images")
    fun deleteAll()

    @Query("SELECT * FROM images")
    fun getAllImages(): Single<List<ImageEntity>>

    @Query("SELECT * FROM images WHERE fingerPrintID = :id")
    fun getAllImagesByFingerprints(id: Long): Single<List<ImageEntity>>
}